### ISSUE TITLE
[Bug] Database data directory permission

### ISSUE THREAD

**Issue by electrical** [goharbor/harbor-helm#876]
If you are reporting a problem, please make sure the following information are provided:

**Expected behavior and actual behavior:**
After upgrading I'm getting the following error of the database instance:

```
FATAL: data directory "/var/lib/postgresql/data/pgdata" has group or world access
DETAIL: Permissions should be u=rwx (0700).
```
Perhaps it would be easy to chmod it before starting up to ensure we have the right permissions?

**Versions:**
Please specify the versions of following systems.

- harbor version: 2.1.3 -> 2.2.0
- Kubernetes: 1.19.7

**Additional context:**

The database instance uses Persisted storage backed by Ceph.

Relevant helm value:
```
    database:
      existingClaim: ""
      storageClass: "rook-ceph-block"
      subPath: "data"
      accessMode: ReadWriteOnce
      size: 1Gi
```

After debugging the issue a bit further I see in the init container it creates the data directory if it doesn't exist and ensures it's created with `0700` permissions.
Unfortunately we don't enforce it once the directory exists.
My suggestion is to replace the `|| true` at the end and convert that into `|| chmod 700 $datadirectory && true`

Note: I first reported this under https://github.com/goharbor/harbor/issues/14354 but was requested to move it here as it's specific to the helm chart deployment.

**Comment by mumrau** [comment]
I am using the same fix as you do for now (thanks, mine was a little more dirty).
It will be important to keep in mind thios issue for future releases, in case the `data-migrator` initContainer disappears, it would be great to replace it with a more classic `data-permissions-ensurer` or something like that.

**Comment by reasonerjt** [comment]
@electrical 
Let me clarify, so in v2.1.3 the perm of the directory was 0700 before upgrade?  and after upgrade it's changed again?

**Comment by ywk253100** [comment]
@electrical @mumrau In Harbor chart 1.6.0, we put the data of database in a subfolder `pgdata` of the mounted PVC to avoid the permission issue reported in #444. The init container `data-migrator` is used to migrate the data into the new subfolder.
The reason why we don't apply the `chmod` command in the init container is described here https://github.com/goharbor/harbor-helm/issues/444#issuecomment-728789140

@mumrau  @Djiit @lotherk @lucioveloso @techgnosis Is the persistent storage is also backed by Ceph for your guys?

We cannot reproduce the issue in our test environment, appreciate if you guys can provide more information about the permission problem

**Comment by mumrau** [comment]
Makes sense.

I am indeed using `ceph` (alongside `rook-ceph`), and it feels like the `0700` permissions aren't persistent accross restart.
A fix making `initContainers` configurable trough the values would probably be great, this chart lacks some tweakables settings in the values (`additionalInitContainers` for every workload makes sense for many installations).

It would also keep working _after_ the `data-migrator` has been removed (which it probably will be removed starting at `harbor 2.3.x` and chart `1.7.x` or maybe even before.

**Comment by Yrok472** [comment]
> @mumrau @Djiit @lotherk @lucioveloso @techgnosis Is the persistent storage is also backed by Ceph for your guys?

I'm using openstack (cinder), same story.

**Comment by Djiit** [comment]
Hey, I'm using default values for this.

**Comment by mediocrematt** [comment]
same issue on a fresh helm deploy. it was working properly, then failed after about 40 minutes. this is an openshift 4.7.0 cluster backed by openshift container storage (ie, rook-ceph).

$ oc get pods
NAME                                           READY   STATUS             RESTARTS   AGE
harbor-harbor-chartmuseum-cd9f8b676-lzn8d      1/1     Running            0          74m
harbor-harbor-core-58b4874cf4-wn2cz            1/1     Running            0          74m
harbor-harbor-database-0                       0/1     CrashLoopBackOff   6          9m54s
harbor-harbor-jobservice-58fb54487d-z8cpg      1/1     Running            0          74m
harbor-harbor-notary-server-7b97fd7b48-lb5md   1/1     Running            1          74m
harbor-harbor-notary-signer-f8f75756b-grzjg    1/1     Running            1          74m
harbor-harbor-portal-6b69b5d9d4-p7pgn          1/1     Running            0          74m
harbor-harbor-redis-0                          1/1     Running            0          74m
harbor-harbor-registry-fbb548475-f4gpv         2/2     Running            0          74m
harbor-harbor-trivy-0                          1/1     Running            0          74m
$ oc logs harbor-harbor-database-0
FATAL:  data directory "/var/lib/postgresql/data/pgdata" has group or world access
DETAIL:  Permissions should be u=rwx (0700).

$ oc debug harbor-harbor-database-0
Starting pod/harbor-harbor-database-0-debug ...
Pod IP: 10.254.1.212
If you don't see a command prompt, try pressing enter.
                                                                                                                                                                                                                                                                               sh-4.4$ ls -lah /var/lib/postgresql/
total 0
drwxr-xr-x 3 root root     18 Feb 18 06:37 .
drwxr-xr-x 1 root root     32 Feb 18 06:38 ..
drwxrwsrwx 3 root postgres  1 Mar  9 21:14 data
sh-4.4$ ls -lah /var/lib/postgresql/data/
total 0
drwxrwsrwx  3 root     postgres  1 Mar  9 21:14 .
drwxr-xr-x  3 root     root     18 Feb 18 06:37 ..
drwxrws--- 19 postgres postgres 24 Mar  9 21:15 pgdata
sh-4.4$ ls -lah /var/lib/postgresql/data/pgdata/
total 31K
drwxrws--- 19 postgres postgres   24 Mar  9 21:15 .
drwxrwsrwx  3 root     postgres    1 Mar  9 21:14 ..
-rw-rw----  1 postgres postgres    4 Mar  9 21:14 PG_VERSION
drwxrws---  8 postgres postgres    6 Mar  9 21:15 base
drwxrws---  2 postgres postgres   54 Mar  9 21:18 global
drwxrws---  2 postgres postgres    1 Mar  9 21:14 pg_clog
drwxrws---  2 postgres postgres    0 Mar  9 21:14 pg_commit_ts
drwxrws---  2 postgres postgres    0 Mar  9 21:14 pg_dynshmem
-rw-rw----  1 postgres postgres 4.4K Mar  9 21:15 pg_hba.conf
-rw-rw----  1 postgres postgres 1.6K Mar  9 21:14 pg_ident.conf
drwxrws---  4 postgres postgres    2 Mar  9 21:14 pg_logical
drwxrws---  4 postgres postgres    2 Mar  9 21:14 pg_multixact
drwxrws---  2 postgres postgres    1 Mar  9 21:15 pg_notify
drwxrws---  2 postgres postgres    0 Mar  9 21:14 pg_replslot
drwxrws---  2 postgres postgres    0 Mar  9 21:14 pg_serial
drwxrws---  2 postgres postgres    0 Mar  9 21:14 pg_snapshots
drwxrws---  2 postgres postgres    0 Mar  9 21:15 pg_stat
drwxrws---  2 postgres postgres    7 Mar  9 22:03 pg_stat_tmp
drwxrws---  2 postgres postgres    1 Mar  9 21:14 pg_subtrans
drwxrws---  2 postgres postgres    0 Mar  9 21:14 pg_tblspc
drwxrws---  2 postgres postgres    0 Mar  9 21:14 pg_twophase
drwxrws---  3 postgres postgres    2 Mar  9 21:14 pg_xlog
-rw-rw----  1 postgres postgres   88 Mar  9 21:14 postgresql.auto.conf
-rw-rw----  1 postgres postgres  22K Mar  9 21:14 postgresql.conf
-rw-rw----  1 postgres postgres   85 Mar  9 21:15 postmaster.opts
-rw-rw----  1 postgres postgres   88 Mar  9 21:15 postmaster.pid

**Comment by ywk253100** [comment]
Most of the breaks are related with Ceph.
Seems that with the `fsGroup`https://github.com/goharbor/harbor-helm/blob/v1.6.0/templates/database/database-ss.yaml#L30, Ceph resets the permission of the mounted volume every restarting. 
If it is, when you correct permission manually and restart the pod, the issue should occur again. 
Correct the permission and remove the `fsGroup` configuration should fix it.

Could you guys help us to verify it if possible?

**Comment by mumrau** [comment]
Correcting the permissions and restarting the pod leads to the same issue. For me the only fix was the one from @electrical ("My suggestion is to replace the `|| true` at the end and convert that into ` || chmod 700 $datadirectory && true`").

I cannot try the fix you suggested (`fsGroup` related) for now sadly, maybe someone that doesn't want to go the `chmod route` can give it a shot.

**Comment by lucioveloso** [comment]
@ywk253100 When do you plan to merge #893 into 1.6.0? I guess it would require releasing a new Helm Chart version, 1.6.1.

**Comment by techgnosis** [comment]
This happens on TKGI 1.10 on vSphere 7U2 backed by VSAN when using Harbor chart 1.6.0 from goharbor
I just booted up my lab and Harbor is broken with this bug

**Comment by ywk253100** [comment]
@lucioveloso The fix has merged into 1.6 branch.

As Harbor 2.2.0 (comsumed by Harbor chart 1.6.0) contains bug which causes high CPU consumption, we will release the Harbor chart 1.6.1 as soon as OSS Harbor 2.2.1 is released
